xclim/indices/run_length.py

```diff
@@ -293,6 +294,160 @@ def last_run(
     return out
 
 
+def run_length_with_date(
+    da: xr.DataArray, window: int, date: str = "07-01", dim: str = "time",
+):
+    """Return the index of the last item of the last run of at least a given length.
```

Comment:
Maybe add a note to clarify that a "run" can includes holes smaller than "window", which is a bit different from the other methods of this file where runs must be at least of length "window" but also *continuous*.  I just realised how confusing that might be, but can't think of another name then "run" for now...

Reply:
I have a suggestion : change "run" to "season". That's what they are used for anyway and it removes the confusion between the definitions of "run" in the other functions of this submodule. Like : `season_end` and `season_length` (Maybe we can assume that te "date" argument is part of the "season" definition and not add it to the name?)

Reply:
This is an interesting point. Maybe we can return to this in another issue. We could also wait until we hear back on the refactoring suggestions.